en/guide/message_schema.md

```diff
@@ -0,0 +1,231 @@
+    They can also be arrays of the other types - e.g. `uint16_t[10]`. 
+  - `name`: Name of the field (used in code).
+  - [enum](#enum) (optional): Name of an `enum` defining possible values of the field (e.g. `MAV_BATTERY_CHARGE_STATE`).
+  - `units` (optional): The units for message `field`s that take numeric values (not enums). These are defined in the [schema](https://github.com/ArduPilot/pymavlink/blob/master/generator/mavschema.xsd) (search on *name="SI_Unit"*)
+  - `display` (optional): This should be set as `display="bitmask"` for bitmask fields (hint to ground station that enum values must be displayed as checkboxes).
+  - `print_format` (optional): TBD.
+  - `default` (optional): TBD.
+- [deprecated](#deprecated) / [wip](#wip) (optional): A tag indicating that the message is deprecated or "work in progress".
+- `extensions` (optional): This self-closing tag is used to indicate that subsequent fields apply to MAVLink 2 only. 
+  - The tag should be used for MAVLink 1 messages only (id < 256) that have been extended in MAVLink 2. 
+
+
+
+
+
+### Others Tags
+
+This section lists a number of tags can be used in a number of other types - e.g. messages and enums.
+
+#### deprecated {#deprecated}
+
+The `<deprecated>` tag can be used in an [enum](#enum), enum [entry](#entry) (value) or [message](#message) to indicate that the item has been superseded. 
+The tag attributes indicates the time of deprecation and the replacement item, while the element may (optionally) contain a string with additional information about the deprecation.
+
+The generator toolchain can be configured to conditionally build messages omitting the `deprecated` entries.
+
+> **Tip** An entity should be marked as deprecated only when the main users have had an opportunity to update to the new method.
+
+As a concrete example, below we see that [SET_MODE](../messages/common.md#SET_MODE) is deprecated and replaced by [MAV_CMD_DO_SET_MODE](../messages/common.md#MAV_CMD_DO_SET_MODE) on `2015-12`.
+```xml
+    <message id="11" name="SET_MODE">
+      <deprecated since="2015-12" replaced_by="MAV_CMD_DO_SET_MODE">Use COMMAND_LONG with MAV_CMD_DO_SET_MODE instead</deprecated>
+```
+
+The `deprecated` attributes are:
+* `since`: Year/month when deprecation started. Format: `YYYY-MM`.
+* `replaced by`: String of entity that supersedes this item.
+
+
+
+#### wip {#wip}
+
+The `<wip>` tag can be used in an enum [entry](#entry) (value) or [message](#message) to indicate that the item is a "work in progress". 
+The element may (optionally) contain a string with additional information about the new item.
+
+The generator toolchain can be configured to conditionally build messages omitting the `wip` entries.
+
+Most commonly, the tag is used as shown:
+```xml
+<wip />
+```
+
+<!--  ref="description" minOccurs="0" -->
```

Comment:
@amilcarlucas Across the schema I see this a number of times. Of course ideally it should be one, but should it not be restricted everywhere to max of one?